x-pack/elastic-agent/pkg/capabilities/input.go

```diff
@@ -0,0 +1,275 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package capabilities
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/transpiler"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/status"
+)
+
+const (
+	inputsKey = "inputs"
+)
+
+func newInputsCapability(log *logger.Logger, rd *ruleDefinitions, reporter status.Reporter) (Capability, error) {
+	if rd == nil {
+		return &multiInputsCapability{log: log, caps: []Capability{}}, nil
+	}
+
+	caps := make([]Capability, 0, len(rd.Capabilities))
+
+	for _, r := range rd.Capabilities {
+		c, err := newInputCapability(log, r, reporter)
+		if err != nil {
+			return nil, err
+		}
+
+		if c != nil {
+			caps = append(caps, c)
+		}
+	}
+
+	return &multiInputsCapability{log: log, caps: caps}, nil
+}
+
+func newInputCapability(log *logger.Logger, r ruler, reporter status.Reporter) (Capability, error) {
+	cap, ok := r.(*inputCapability)
+	if !ok {
+		return nil, nil
+	}
+
+	cap.log = log
+	cap.reporter = reporter
+	return cap, nil
+}
+
+type inputCapability struct {
+	log      *logger.Logger
+	reporter status.Reporter
+	Name     string `json:"name,omitempty" yaml:"name,omitempty"`
+	Type     string `json:"rule" yaml:"rule"`
+	Input    string `json:"input" yaml:"input"`
+}
+
+func (c *inputCapability) Apply(in interface{}) (bool, interface{}) {
```

Comment:
i delegated this responsibility to the latest caller. in case inputCapability is used outside of multiInput i wanted to have checks in place. 
it might be redundant but it's not run that often